My profile and auth

Svyatoslav-M opened the pull request:
Exercise 1
89a98b7: Added "My profile" screen
e72e4c1: 1. Drawables for buttons and social networks icons 2. Style for rounded profile icon
d010212: Other changes (in manifest, in gradle, colors, strings, fonts, pics)

Exercise 3
3dc705a: Was added authorization screen. Made email and password validation. Made navigation from AuthActivity to MainActivity with profile page.
31d88ed: Other changes (in manifest, gradle, strings, pics, drawables, themes)

After review:
e4f9b8c: Was added .idea  to .gitignore
a180c1f: 1. Was deleted Compose from build.gradle 3. Was changed minSdk to 26 2. Was changed marginTop for buttons in activity_auth.xml
35c582c: Was deleted my_photo_foreground and my_photo_round
28b026df: 1. Changes in activity_auth.xml from review 2. Changes in colors.xml from review 3. Changes in strings.xml from review 4. Was created dimens.xml 5. Was created custom checkbox 6. Was changed buttons color
7aa13645: Was changed my profile activity
e3789aca: 1. Was changed MyProfileActivity.kt, AuthActivity.kt, NameParser.kt, FieldsChecker.kt 2. Was added Constants object 3. Was changed manifest 4. Other small changes

Svyatoslav-M committed:
Added "My profile" screen

Svyatoslav-M committed:
1. Drawables for buttons and social networks icons 2. Style for rounded profile icon

Svyatoslav-M committed:
Other changes (in manifest, in gradle, colors, strings, fonts, pics)

Svyatoslav-M committed:
Was added authorization screen. Made email and password validation. Made navigation from AuthActivity to MainActivity with profile page.

Svyatoslav-M committed:
Other changes (in manifest, gradle, strings, pics, drawables, themes)

Svyatoslav-M committed:
Was added colors for day and night mode; Was changed colors name in buttons xml. Was changed one line in themes.xml.

Svyatoslav-M committed:
Revert "Was added colors for day and night mode; Was changed colors name in buttons xml. Was changed one line in themes.xml."

This reverts commit 6515a37417a342064e4c34544d3202c880ef738e.

Svyatoslav-M committed:
Revert "Revert "Was added colors for day and night mode; Was changed colors name in buttons xml. Was changed one line in themes.xml.""

This reverts commit 0fde1ed85794e344ca4cac161ae5284fb091eab7.

Svyatoslav-M committed:
Was added colors for day and night mode; Was changed colors name in buttons xml. Was changed one line in themes.xml.

Pl-Alex commented:
I noticed the project includes Jetpack Compose, but we only use Views in this course. To simplify the project, consider removing Compose.

vikmanz commented:
Common:
- Folder .idea can be added to .gitignore
- build.gradle -> minSdk -> 26
- build.gradle -> As Pi-Alex mentioned, it may be worth removing Compose if the project is on View. When a project uses both technologies, it weighs more and is slower.

vikmanz commented on app/src/main/res/values/colors.xml:
```diff
@@ -7,4 +7,7 @@
     <color name="teal_700">#FF018786</color>
     <color name="black">#FF000000</color>
     <color name="white">#FFFFFFFF</color>
```
Don't use color names in your titles. The color may change from orange to green, but the name will remain “orange.” It's better to name it for its intended purpose.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,65 @@
+    <!-- Strings used for fragments for navigation -->
+    <string name="first_fragment_label">First Fragment</string>
+    <string name="second_fragment_label">Second Fragment</string>
+    <string name="next">Next</string>
```
It's the same with color names. It is better to name all strings for its intended purpose.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,65 @@
+    <string name="next">Next</string>
+    <string name="previous">Previous</string>
+
+    <string name="lorem_ipsum">
```
Is it being used?

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        app:layout_constraintGuide_begin="20dp" />
```
Guidelines are usually used with percentage values rather than hard-coded values.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+    tools:context=".AuthActivity.AuthActivity">
+
+    <androidx.constraintlayout.widget.Guideline
+        android:id="@+id/guideline"
```
Naming should be a bit more understandable. What is this guideline?

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        android:layout_marginTop="120dp"
```
All dimensions (dp/sp) must be extracted to dimens.xml

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        app:layout_constraintGuide_end="20dp" />
+
+    <TextView
+        android:id="@+id/textView"
```
All naming should be a bit more understandable. And add type of view tag at start of name.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:id="@+id/textView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
```
Why not from guidelines?

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:orientation="vertical"
+        app:layout_constraintGuide_end="20dp" />
+
+    <TextView
```
Use AppCompat versions of elements. AppCompatTextView

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:textColor="@color/white"
+        android:textSize="16sp" />
+
+    <EditText
```
Use TextInputLayout. And errors not as in design.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:inputType="textEmailAddress"
+        android:textColor="@color/white" />
+
+    <TextView
```
Rearrange tags

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:inputType="textPassword"
+        android:textColor="@color/white" />
+
+    <CheckBox
```
AppCompatCheckBox. And if you click it, the checkbox is not the same as in the design.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:textColor="@color/profile_gray"
+        android:buttonTint="@color/white" />
+
+    <LinearLayout
```
Layout in layout - bad idea. In Constraint layout you can use Flow, as alternative of Linear layout.

Svyatoslav-M replied:
I tried to make with Flow, but didn't understand, how to make white background

vikmanz replied:
for white background use simple view in main constraint

Svyatoslav-M replied:
Thank you! That's working. Commit 61e9337

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:orientation="horizontal"
+        android:padding="8dp">
+
+        <ImageView
```
AppCompatImageView

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:fontFamily="@font/open_sans_medium"
+        android:text="@string/by_clicking_register_you_agree_to_our_terms_and_conditions"
```
Not like in design (2 lines)

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_auth.xml:
```diff
@@ -0,0 +1,208 @@
+        android:layout_marginEnd="32dp"
+        android:fontFamily="@font/open_sans_medium"
+        android:text="@string/sign_in"
+        android:textColor="@color/orange"
```
Not like in design.

Svyatoslav-M replied:
I've made the changes. Commit 28b026d

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+    tools:context=".MyProfileActivity.MyProfileActivity"
+    android:orientation="vertical">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
ConstraintLayout in ConstraintLayout - VERY bad idea. Performance crying.

Svyatoslav-M replied:
I've made the changes. Commit 7aa13645

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            android:textSize="15sp"/>
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```
Second ConstraintLayout in ConstraintLayout - VERY VERY bad idea. Performance is very poor. For BG you can use simple View.

Svyatoslav-M replied:
I've made the changes. Commit 7aa13645

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            android:typeface="serif" />
+
+        <TextView
+            android:id="@+id/nameTextView"
```
Add type of view tag at start of name. This will simplify the search in code.

Svyatoslav-M replied:
I've made the changes. Commit 7aa13645

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            android:layout_height="wrap_content"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/profileImageView"
```
Use Chain for Foto, name, career and home address.

Svyatoslav-M replied:
I tryed to use Chain, but didn't understand how to make different spacing between objects like in design.

vikmanz replied:
Chain - packed + margin for element if need

Svyatoslav-M replied:
Yeah, I know that, but I thought there was some more polite way to do it. I tryed bias, but that's didn't work or I didn't understand how to work with it

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            app:layout_constraintStart_toStartOf="@+id/careerTextView"
+            app:layout_constraintTop_toBottomOf="@id/careerTextView"
+            android:layout_marginTop="16dp"
+            android:layout_marginBottom="45dp"
```
If it is processed into a chain, you don't need it.

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+        app:layout_constraintTop_toBottomOf="@id/topDarkFrame"
+        app:layout_constraintVertical_weight=".5">
+
+        <RelativeLayout
```
Don't use layout in layout in layout. More levels of nesting means less productivity. And use Chain for social icons.

Svyatoslav-M replied:
I've made the changes. Commit 7aa1364

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            android:background="@drawable/border_social_network"
+            android:padding="5dp">
+
+            <ImageView
```
AppCompatImageView

Svyatoslav-M replied:
I've made the changes. Commit 7aa1364

vikmanz commented on app/src/main/res/layout/activity_my_profile.xml:
```diff
@@ -0,0 +1,225 @@
+            android:text="@string/edit_profile"
+            android:textAllCaps="false"/>
+
+        <Button
```
AppCompatButton. And corners not same as in design

Svyatoslav-M replied:
I've made the changes. Commit 7aa1364

vikmanz commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,19 +10,29 @@
+        android:theme="@style/Theme.AppCompat.Light"
+
         tools:targetApi="31">
         <activity
```
Use only 2 activities!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+package com.example.homework_1.AuthActivity
```
fix package

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+ */
+class AuthActivity : AppCompatActivity() {
+
+    private var editTextEmail : EditText? = null
```
If you need it (all these variables) - create variables local.

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_auth)
+
+        editTextEmail = findViewById(R.id.editTextTextEmailAddress)
```
findViewById is old approach. Use binding instead!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+
+        fieldsChecker = FieldsChecker()
+
+        buttonRegister.setOnClickListener {
```
Extract setListeners()

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+            if (isAllFieldsCorrect) {
+                val intent = Intent(this@AuthActivity, MainActivity::class.java)
+                intent.putExtra("email", editTextEmail!!.text.toString())
+                startActivity(intent)
```
Decompose method

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+
+            if (isAllFieldsCorrect) {
+                val intent = Intent(this@AuthActivity, MainActivity::class.java)
+                intent.putExtra("email", editTextEmail!!.text.toString())
```
String keys must be a constants!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/AuthActivity.kt:
```diff
@@ -0,0 +1,46 @@
+
+            if (isAllFieldsCorrect) {
+                val intent = Intent(this@AuthActivity, MainActivity::class.java)
+                intent.putExtra("email", editTextEmail!!.text.toString())
```
NOT SAFE! .?

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+package com.example.homework_1.AuthActivity
```
package

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Class for checking email and passwords fields.
+ * */
+class FieldsChecker {
```
Object. Not class

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+     * @param editTextPassword EditText with password
+     * @return True or False
+     * */
+    fun checkAllFields(editTextEmail: EditText?, editTextPassword: EditText?): Boolean {
```
Send views to other classes is bad idea. Send only text! Return error message.

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+     * @param editTextPassword EditText with password
+     * @return True or False
+     * */
+    fun checkAllFields(editTextEmail: EditText?, editTextPassword: EditText?): Boolean {
```
Split to 2 different methods.

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+     * */
+    fun checkAllFields(editTextEmail: EditText?, editTextPassword: EditText?): Boolean {
+        //Regex for email.
+        val emailRegex = Regex("^[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}\$")
```
Java Patterns.Email

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+        //Regex for password. Minimum eight characters, at least one letter and one number
+        val passwordRegex = Regex("^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}\$")
+
+        if (editTextEmail!!.length() == 0) {
```
NOT SAFE!!!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+        val passwordRegex = Regex("^(?=.*[A-Za-z])(?=.*\\d)[A-Za-z\\d]{8,}\$")
+
+        if (editTextEmail!!.length() == 0) {
+            editTextEmail!!.error = "This field is required"
```
All strings -> to values\strings.xml

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/AuthActivity/FieldsChecker.kt:
```diff
@@ -0,0 +1,42 @@
+            return false
+        }
+        if (!passwordRegex.matches(editTextPassword!!.text.toString())) {
+            editTextPassword!!.error = "Password could contains at least:\n\t8 characters" +
```
8 must be const and passed to get string from resource method (%1$d)

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MainActivity.kt:
```diff
@@ -0,0 +1,26 @@
+
+        parser = NameParser()
+
+        val email = intent.getStringExtra("email")
```
Const string key from AuthActivity. 
And decompose setting values to separate method.

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MainActivity.kt:
```diff
@@ -0,0 +1,26 @@
+        val email = intent.getStringExtra("email")
+
+        name = findViewById(R.id.nameTextView)
+        name!!.text = parser.parseNameSurname(email!!)
```
Don't use not-safe access!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MainActivity.kt:
```diff
@@ -0,0 +1,26 @@
+package com.example.homework_1.MyProfileActivity
```
package. And why MainActivity inside MyProfileActivity?

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MyProfileActivity.kt:
```diff
@@ -0,0 +1,12 @@
+import android.os.Bundle
+import com.example.homework_1.R
+
+class MyProfileActivity : AppCompatActivity() {
```
Delete this?

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MainActivity.kt:
```diff
@@ -0,0 +1,26 @@
+/**
+ * Class for my profile activity.
+ */
+class MainActivity : AppCompatActivity() {
```
Maybe it is MyProfileActivity?

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/MainActivity.kt:
```diff
@@ -0,0 +1,26 @@
+ */
+class MainActivity : AppCompatActivity() {
+
+    var name: TextView? = null
```
local vars

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/NameParser.kt:
```diff
@@ -0,0 +1,39 @@
+package com.example.homework_1.MyProfileActivity
```
package

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/NameParser.kt:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Name and surname parser class.
+ */
+class NameParser {
```
Object. Not a class

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/NameParser.kt:
```diff
@@ -0,0 +1,39 @@
+        var surname = ""
+        var changer = false
+
+        for (ch in email) {
```
Use string methods!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

vikmanz commented on app/src/main/java/com/example/homework_1/MyProfileActivity/NameParser.kt:
```diff
@@ -0,0 +1,39 @@
+        if (surname == "") {
+            return name
+        }
+        return "${name.capitalize()} ${surname.capitalize()}"
```
Deprecated methods!

Svyatoslav-M replied:
I've made the changes. Commit e3789aca

Svyatoslav-M committed:
Update .gitignore

Was added idea to gitignore

Svyatoslav-M committed:
1. Was deleted Compose from build.gradle
2. Was changed marginTop for buttons in activity_auth.xml
3. Was deleted my_photo_foreground and my_photo_round

Svyatoslav-M committed:
Merge branch 'dev' of https://github.com/Svyatoslav-M/android_lvl1 into dev

Svyatoslav-M committed:
Was deleted my_photo_foreground and my_photo_round

Svyatoslav-M commented:
> Common:
> 
> * Folder .idea can be added to .gitignore
> * build.gradle -> minSdk -> 26
> * build.gradle -> As Pi-Alex mentioned, it may be worth removing Compose if the project is on View. When a project uses both technologies, it weighs more and is slower.
> * No need for my_photo_foreground and my_photo_round

Thank you! I've made the changes

Svyatoslav-M commented:
> I've noticed that the project includes Jetpack Compose, but its features are not being utilized. Since we are only using Views in this course, it might be better to remove Compose dependencies to keep the project simpler

Thank you! I've made the changes

Svyatoslav-M committed:
1. Changes in activity_auth.xml from review
2. Changes in colors.xml from review
3. Changes in strings.xml from review
4. Was created dimens.xml
5. Was created custom checkbox
6. Was changed buttons color

Svyatoslav-M committed:
1. Changes in activity_auth.xml from review
2. Changes in colors.xml from review
3. Changes in strings.xml from review
4. Was created dimens.xml
5. Was created custom checkbox
6. Was changed buttons color

Svyatoslav-M committed:
Was changed google button from LinearLayout to Flow

Svyatoslav-M committed:
Some changes

Svyatoslav-M committed:
Merge branch 'dev' of https://github.com/Svyatoslav-M/android_lvl1 into dev

# Conflicts:
#	app/src/main/res/layout/activity_auth.xml
#	app/src/main/res/values/dimens.xml

Svyatoslav-M committed:
Was changed my profile activity

Svyatoslav-M committed:
1. Was changed MyProfileActivity.kt, AuthActivity.kt, NameParser.kt, FieldsChecker.kt
2. Was added Constants object
3. Was changed manifest
4. Other small changes

Svyatoslav-M committed:
Delete MyProfileActivity.kt, MainActivity was renamed to this

Svyatoslav-M committed:
1. Was created colors.xml for night theme and a little bit eddied colors.xml for day theme
2. Was created strings.xml for en and uk languages
3. Change one line in activity_my_profile.xml

Svyatoslav-M committed:
Was added email and password save and autologin with SharedPreferences

Svyatoslav-M committed:
Deleted "activity" word in strings.xml

Svyatoslav-M committed:
Made changes after review.

Svyatoslav-M committed:
small change

Svyatoslav-M committed:
Merge pull request #5 from Svyatoslav-M/save_data_with_sharedprederences

Save data with SharedPreferences

Svyatoslav-M committed:
Merge pull request #4 from Svyatoslav-M/night_theme_and_languages

Added night theme and languages
